Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -538,24 +538,26 @@ private bool IsWarning(string line)
         {
             internal readonly Predicate<string> LinePredicate;
             internal readonly Regex Regex;
+            internal readonly string GroupName;
 
             internal OutputFilter(Predicate<string> linePredicate)
-                : this(linePredicate, null)
+                : this(linePredicate, null, null)
             {
                 Contract.Requires(linePredicate != null);
             }
 
-            internal OutputFilter(Regex regex)
-                : this(null, regex)
+            internal OutputFilter(Regex regex, string groupName = null)
+                : this(null, regex, groupName)
             {
                 Contract.Requires(regex != null);
             }
 
-            private OutputFilter(Predicate<string> linePredicate, Regex regex)
+            private OutputFilter(Predicate<string> linePredicate, Regex regex, string groupName)
             {
                 Contract.Requires(linePredicate != null || regex != null);
                 LinePredicate = linePredicate;
                 Regex = regex;
+                GroupName = groupName ?? ErrorMessageGroupName;

```

Comment:
>ErrorMessageGroupName [](start = 41, length = 21)

I think the caller of `OutputFilter`'s constructor is responsible for passing this group name. I see only one caller, so it's should be easy to update.

Reply:
Is the ask to make GroupName an explicitly required parameter (and then update all the callers to pass that)?